src/Iot.Device.Bindings/Iot.Device.Bindings.csproj

```diff
@@ -28,6 +28,7 @@
     <ProjectReference Include="$(MainLibraryPath)System.Device.Gpio.csproj" />
     <PackageReference Include="UnitsNet" Version="$(UnitsNetPackageVersion)" />
     <PackageReference Include="System.Management" Version="$(SystemManagementPackageVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="$(MicrosoftExtensionsVersion)" />
```

Comment:
I check the source code and we will have hard time to make this working for nano as it's using various generics <T> but we may just rewrite using your initial code couple of high level wrappers which make the tricks for nano

Reply:
It looks weird, but I think it shouldn't be such a big deal, since their usage of generics serves - as far as I have seen - no real purpose. They mostly use generics to forward the class types of the logging sources. So the generic arguments should be easily replaced by an additional string or type argument.

Reply:
Note that the implementation I use tries to reduce the visible generics from the imported library. If you redefine the ILogger interface for nano without generics, there will be very few changes required.

Reply:
Yes, we will manage a way to adjust. Not too stressed about it!